File: system-test/dotnet/Dsl.Core/UseCase/External/Clock/ClockDsl.cs
using Dsl.Core.External.Clock.UseCases;
using Driver.Port.External.Clock;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Clock;

public class ClockDsl : IDisposable
{
    private readonly IClockDriver _driver;
    private readonly UseCaseContext _context;
    private bool _disposed;

    public ClockDsl(IClockDriver driver, UseCaseContext context)
    {
        _driver = driver;
        _context = context;
    }

    public void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool disposing)
    {
        if (_disposed) return;
        if (disposing)
            _driver?.Dispose();
        _disposed = true;
    }

    public GoToClock GoToClock() => new(_driver, _context);

    public ReturnsTime ReturnsTime() => new(_driver, _context);

    public GetTime GetTime() => new(_driver, _context);
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/core/usecase/external/clock/ClockDsl.java
package com.mycompany.myshop.testkit.dsl.core.usecase.external.clock;

import com.mycompany.myshop.testkit.driver.port.external.clock.ClockDriver;
import com.mycompany.myshop.testkit.dsl.core.usecase.external.clock.usecases.GetTime;
import com.mycompany.myshop.testkit.dsl.core.usecase.external.clock.usecases.GoToClock;
import com.mycompany.myshop.testkit.dsl.core.usecase.external.clock.usecases.ReturnsTime;
import com.mycompany.myshop.testkit.common.Closer;
import com.mycompany.myshop.testkit.dsl.core.shared.UseCaseContext;

public class ClockDsl implements AutoCloseable {
    protected final ClockDriver driver;
    protected final UseCaseContext context;

    public ClockDsl(ClockDriver driver, UseCaseContext context) {
        this.driver = driver;
        this.context = context;
    }

    @Override
    public void close() {
        Closer.close(driver);
    }

    public GoToClock goToClock() {
        return new GoToClock(driver, context);
    }

    public ReturnsTime returnsTime() {
        return new ReturnsTime(driver, context);
    }

    public GetTime getTime() {
        return new GetTime(driver, context);
    }
}
